apps/meteor/app/cloud/server/functions/syncWorkspace/fetchWorkspaceSyncPayload.ts

```diff
@@ -0,0 +1,46 @@
+import type { Cloud, Serialized } from '@rocket.chat/core-typings';
+import { serverFetch as fetch } from '@rocket.chat/server-fetch';
+import { v, compile } from 'suretype';
+
+import { CloudWorkspaceConnectionError } from '../../../../../lib/errors/CloudWorkspaceConnectionError';
+import { settings } from '../../../../settings/server';
+
+const workspaceSyncPayloadSchema = v.object({
+	workspaceId: v.string().required(),
+	publicKey: v.string(),
+	license: v.string().required(),
+});
+
+const assertWorkspaceSyncPayload = compile(workspaceSyncPayloadSchema);
+
+export async function fetchWorkspaceSyncPayload({
+	token,
+	data,
+}: {
+	token: string;
+	data: Cloud.WorkspaceSyncRequestPayload;
+}): Promise<Serialized<Cloud.WorkspaceSyncResponse>> {
+	const workspaceRegistrationClientUri = settings.get<string>('Cloud_Workspace_Registration_Client_Uri');
+	const response = await fetch(`${workspaceRegistrationClientUri}/sync`, {
+		method: 'POST',
+		headers: {
+			Authorization: `Bearer ${token}`,
```

Comment:
@lucas-a-pelegrino 
Just to be clear, I'm not against refactors, but I'm against them during feature development as it makes it difficult to understand the diffs.

Reply:
Sorry about that! I'll keep it in mind next time.

The reason I've refactored this function into a new file was so it was easier to test it.